api/web3server.go

```diff
@@ -459,33 +459,51 @@ func (svr *web3Handler) sendRawTransaction(in *gjson.Result) (interface{}, error
 	}
 	// parse raw data string from json request
 	var (
-		cs       = svr.coreService
-		tx       *types.Transaction
-		encoding iotextypes.Encoding
-		sig      []byte
-		pubkey   crypto.PublicKey
-		err      error
+		cs        = svr.coreService
+		rawString = dataStr.String()
+		req       *iotextypes.Action
 	)
-	tx, err = action.DecodeEtherTx(dataStr.String())
-	if err != nil {
-		return nil, err
-	}
-	if tx.Protected() && tx.ChainId().Uint64() != uint64(cs.EVMNetworkID()) {
-		return nil, errors.Wrapf(errInvalidEvmChainID, "expect chainID = %d, got %d", cs.EVMNetworkID(), tx.ChainId().Uint64())
-	}
-	encoding, sig, pubkey, err = action.ExtractTypeSigPubkey(tx)
-	if err != nil {
-		return nil, err
-	}
-	elp, err := svr.ethTxToEnvelope(tx)
-	if err != nil {
-		return nil, err
-	}
-	req := &iotextypes.Action{
-		Core:         elp.Proto(),
-		SenderPubKey: pubkey.Bytes(),
-		Signature:    sig,
-		Encoding:     encoding,
+	if g := cs.Genesis(); g.IsToBeEnabled(cs.TipHeight()) {
+		if strings.HasPrefix(rawString, "0x") || strings.HasPrefix(rawString, "0X") {
+			rawString = rawString[2:]
+		}
+		rawBytes, err := hex.DecodeString(rawString)
+		if err != nil {
+			return nil, err
+		}
+		req = &iotextypes.Action{
+			Core: &iotextypes.ActionCore{
+				ChainID: cs.ChainID(),
+				Action: &iotextypes.ActionCore_EvmTxContainer{
+					EvmTxContainer: &iotextypes.EvmTxContainer{
+						Raw: rawBytes,
+					},
+				},
+			},
+			Encoding: iotextypes.Encoding_ETHEREUM_CONTAINER,
+		}
+	} else {
+		tx, err := action.DecodeEtherTx(rawString)
+		if err != nil {
+			return nil, err
+		}
+		if tx.Protected() && tx.ChainId().Uint64() != uint64(cs.EVMNetworkID()) {
+			return nil, errors.Wrapf(errInvalidEvmChainID, "expect chainID = %d, got %d", cs.EVMNetworkID(), tx.ChainId().Uint64())
+		}
+		encoding, sig, pubkey, err := action.ExtractTypeSigPubkey(tx)
+		if err != nil {
+			return nil, err
+		}
+		elp, err := svr.ethTxToEnvelope(tx)
```

Comment:
`elp` is the only thing we need to determine inside of workingset

Reply:
yes, that's right. So what is your suggestion? 
this is in `else` part and will be deleted after hard-fork

Reply:
To construct an iotextypes.Action, the following fields need to be set:
```
		Core:         ...,
		SenderPubKey: pubkey.Bytes(),
		Signature:    sig,
		Encoding:     encoding,
```
and `Core` is the only place which cannot be determined so far, but we can still fill the other fields.
In this file, which is an API entry point, it doesn't hurt to do https://github.com/iotexproject/iotex-core/pull/4278/files#r1616309346 for the `else` case